rhizome/kubernetes/bin/init-cluster

```diff
@@ -0,0 +1,44 @@
+#!/bin/env ruby
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+
+if ARGV.count != 6
+  fail "Wrong number of arguments. Expected 6, Given #{ARGV.count}"
+end
+
+cluster_name = ARGV[0]
+load_balancer_host_name = ARGV[1]
+port = ARGV[2]
+private_subnet_cidr4 = ARGV[3]
+private_subnet_cidr6 = ARGV[4]
+vm_cidr = ARGV[5]
+
+config = <<YAML
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: ClusterConfiguration
+clusterName: #{cluster_name}
+kubernetesVersion: stable
+controlPlaneEndpoint: #{load_balancer_host_name}:#{port}
+apiServer:
+  certSANs:
+    - #{load_balancer_host_name}
+networking:
+  podSubnet: "#{private_subnet_cidr4},#{private_subnet_cidr6}"
+  dualStack: true
+controllerManager:
+  extraArgs:
+    allocate-node-cidrs: "false"
+nodeRegistration:
+  kubeletExtraArgs:
+    pod-cidr: "#{vm_cidr}"
+etcd:
+  local:
+    dataDir: "/var/lib/etcd"
+YAML
+
```

Comment:
should we consider a yaml dump here? this is asking for quoting issues. I know vmsetup does it this way (and I did it) for cloud-init but it's not to be repeated in a more civilized era

Reply:
![image](https://github.com/user-attachments/assets/86f7ff33-86ac-43c5-a1ef-70a978386199)